packages/blocks/src/embed-block/image/image-block.ts

```diff
@@ -118,15 +119,60 @@ export class ImageBlockComponent extends NonShadowLitElement {
   @state()
   private _source!: string;
 
+  @state()
+  private _imageReady: Disposable = {
+    dispose: () => {
+      return;
+    },
+  };
+
+  @state()
+  private _imageState: 'waitUploaded' | 'loading' | 'ready' = 'loading';
+
+  private waitImageUploaded() {
+    return new Promise<void>(resolve => {
+      const disposeSignal = this.model.page.awarenessStore.signals.update.on(
+        () => {
+          const isBlobUploading =
+            this.model.page.awarenessStore.isBlobUploading(this.model.sourceId);
+          if (!isBlobUploading) {
+            resolve();
+          }
+        }
+      );
+
+      this._imageReady.dispose = () => {
+        disposeSignal.dispose();
+        resolve();
+      };
+    });
+  }
+
   async firstUpdated() {
     this.model.propsUpdated.on(() => this.requestUpdate());
     this.model.childrenUpdated.on(() => this.requestUpdate());
     // exclude padding and border width
     const { width, height } = this.model;
     const storage = await this.model.page.blobs;
     assertExists(storage);
+
+    // FIXME: We cannot guarantee the synchronization order of ydoc and awareness
+    // If ydoc update first, we cannot get upload state for awareness
+    await sleep(1000);
```

Comment:
little weird here. I think we should use something like `event emitter` here

Reply:
Using `event emitter` is also wired. If we use `event emitter`, image block could not figure out when to load the data. Otherwise, image block shows a 404 state, then turns to an upload state.